### ISSUE TITLE
Upgrade to node-fetch v3 to avoid punycode deprecation warning

### ISSUE THREAD

**Issue by andrews05** [getsentry/sentry-cli#1810]
On node 21, using sentry cli shows a warning:
```DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.```

Here's the dependency stack leading to punycode:
```
@sentry/cli@npm:2.21.2
node-fetch@npm:2.7.0
whatwg-url@npm:5.0.0
tr46@npm:0.0.3
punycode
```

Upgrading to node-fetch v3 should resolve this. I believe this will require bumping minimum node to v12, hopefully that won't be issue.

**Comment by loewenheim** [comment]
Hi, thank you for the report. I'll put it on the backlog for now.

**Comment by emmgfx** [comment]
Any news about this? Node 21 was released 5 months ago :(

**Comment by szokeasaurusrex** [comment]
@emmgfx Since implementing this change would require increasing the minimum Node version, this change likely should wait until we release a new major version of the CLI.

**Comment by mo** [comment]
node v22 is very soon LTS and this punycode warning is really annoying... can you please upgrade to node-fetch 3.x? since other very common packages like jsdom and jest has already upgraded to node-fetch 3.x a very large amount of apps cannot support node v10 anyway.. of course if there is a way to silence the warning this would also be a good solution

**Comment by mo** [comment]
running with NODE_OPTIONS="--disable-warning DEP0040" will disable the deprecation warning for punycode

**Comment by szokeasaurusrex** [comment]
Hi @mo, as I mentioned in a [previous comment](https://github.com/getsentry/sentry-cli/issues/1810#issuecomment-2014733786), this likely requires us to release a major version of Sentry CLI because it would be a breaking change. We have no timeline on when we would plan to do this.

> running with NODE_OPTIONS="--disable-warning DEP0040" will disable the deprecation warning for punycode

If I understand correctly, you are able to disable the warning and work around this issue, right?

**Comment by mo** [comment]
Yes, suppressing the warning with --disable-warning DEP0040 works well in my case.

**Comment by khallmark** [comment]
> requires us to release a major version of Sentry CLI

@szokeasaurusrex - There is no reason a major release can't be a single commit with a single breaking change. Don't let other features hold back keeping packages up to date. Make the 3.x version 4.x

**Comment by szokeasaurusrex** [comment]
@khallmark thank you for the feedback

**Comment by mbrevda** [comment]
or just use node's native fetch()if [targeting v18+](https://nodejs.org/docs/v22.12.0/api/globals.html#fetch:~:text=No%20longer%20experimental.-,v18.0.0,-No%20longer%20behind)

**Comment by khallmark** [comment]
@szokeasaurusrex It's time to drop support for EOL version of Node, update to the proper fetch API, remove this package, and make a new release. Out of all the major cloud cli I use, this is the only one that still supports ancient node versions. For funsies I had Cursor replace node-fetch with native fetch using Gemini Pro 2.5 Max for about $2. YMMV.

**Comment by KrzysztofZawisla** [comment]
bump

**Comment by szokeasaurusrex** [comment]
@JPeer264 this should be unblocked by getsentry/sentry-cli#2986, would you like to implement this issue as well? Otherwise, I can also take it
